src/main/java/net/sf/jsqlparser/parser/ParserKeywordsUtils.java

```diff
@@ -0,0 +1,429 @@
+/*-
+ * #%L
+ * JSQLParser library
+ * %%
+ * Copyright (C) 2004 - 2021 JSQLParser
+ * %%
+ * Dual licensed under GNU LGPL 2.1 or Apache License 2.0
+ * #L%
+ */
+package net.sf.jsqlparser.parser;
+
+import org.javacc.jjtree.JJTree;
+import org.javacc.parser.JavaCCGlobals;
+import org.javacc.parser.JavaCCParser;
+import org.javacc.parser.RCharacterList;
+import org.javacc.parser.RChoice;
+import org.javacc.parser.RJustName;
+import org.javacc.parser.ROneOrMore;
+import org.javacc.parser.RSequence;
+import org.javacc.parser.RStringLiteral;
+import org.javacc.parser.RZeroOrMore;
+import org.javacc.parser.RZeroOrOne;
+import org.javacc.parser.RegularExpression;
+import org.javacc.parser.Semanticize;
+import org.javacc.parser.Token;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InvalidClassException;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ParserKeywordsUtils {
+    public final static CharsetEncoder CHARSET_ENCODER = StandardCharsets.US_ASCII.newEncoder();
+
+    public final static int RESTRICTED_FUNCTION = 1;
+    public final static int RESTRICTED_SCHEMA = 2;
+    public final static int RESTRICTED_TABLE = 4;
+    public final static int RESTRICTED_COLUMN = 8;
+    public final static int RESTRICTED_EXPRESSION = 16;
+    public final static int RESTRICTED_ALIAS = 32;
+    public final static int RESTRICTED_SQL2016 = 64;
+
+    public final static int RESTRICTED_JSQLPARSER =  128
+            | RESTRICTED_FUNCTION
+            | RESTRICTED_SCHEMA
+            | RESTRICTED_TABLE
+            | RESTRICTED_COLUMN
+            | RESTRICTED_EXPRESSION
+            | RESTRICTED_ALIAS
+            | RESTRICTED_SQL2016;
+
+
+    @SuppressWarnings({"PMD.ExcessiveMethodLength"})
+    public static List<String> getReservedKeywords(int restriction) {
+        // Classification follows http://www.h2database.com/html/advanced.html#keywords
+        Object[][] ALL_RESERVED_KEYWORDS = {
```

Comment:
Couldn't thoses lists directly derived from the grammar? At least RESTRICTED_ALIAS?

Reply:
It is the other way around: This List will define, how we rewrite `RelObjectName...`. This List will be defined manually and everything else follows it.